### DISCUSSION TITLE
Give option for end-user to have customized column names

### DISCUSSION THREAD

**Discussion by rahul-dt** [airbytehq/airbyte#36064] [Connector Ideas and Features]
Currently, Airbyte doesn't have an option to have customized column names for tables. If due to some unforeseen circumstances, its required to switch between Airbyte and say Fivetran for data syncups, column names don't match up and thence, quite a few changes are required in the codebase to align things. 
Fivetran does column naming according to [this](https://fivetran.com/docs/getting-started/core-concepts#namingconventions)

Tried uploading data to the BigQuery warehouse and source being Google Sheets, the column names in my Google Sheets were -  **col__1**, **_col2**, **col 3**, **col - 4**
Leaving some examples below as to what is desired :- 

*  The column name **col__1** remained as **col__1**; Fivetran would have renamed it as **col_1**
* The column name **col - 4** was renamed as **col___4**; Fivetran would have renamed it as **col_4**

Basically, the idea is to provide the end-user with an option so that they can customize their column-names, albeit following the Airbyte configurations.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368163410929) by [Unito](https://www.unito.io)

**Comment by marcosmarxm** [comment]
That's a great idea @rahulpadhy7

**Comment by ChristopheDuong** [comment]
Do you think this [feature](https://github.com/airbytehq/airbyte/issues/2959) would be sufficient and acceptable to solve your issue with column names?

**Comment by rahul-dt** [comment]
> Do you think this [feature](https://github.com/airbytehq/airbyte/issues/2959) would be sufficient and acceptable to solve your issue with column names?

Hi @ChristopheDuong , more precisely, [this feature](https://github.com/airbytehq/airbyte/issues/3209) should solve where I'm stuck.

**Comment by ChristopheDuong** [comment]
> > Do you think this [feature](https://github.com/airbytehq/airbyte/issues/2959) would be sufficient and acceptable to solve your issue with column names?
> 
> Hi @ChristopheDuong , more precisely, [this feature](https://github.com/airbytehq/airbyte/issues/3209) should solve where I'm stuck.

the issue you linked will also be worked on soon, but it will simply port the "boolean" setting for toggling normalization on and off to the connection/sync page.

**Comment by rahul-dt** [comment]
> > > Do you think this [feature](https://github.com/airbytehq/airbyte/issues/2959) would be sufficient and acceptable to solve your issue with column names?
> > 
> > 
> > Hi @ChristopheDuong , more precisely, [this feature](https://github.com/airbytehq/airbyte/issues/3209) should solve where I'm stuck.
> 
> the issue you linked will also be worked on soon, but it will simply port the "boolean" setting for toggling normalization on and off to the connection/sync page.

Hey @ChristopheDuong , you're right. Basically the need is to have some sort of access for the end-user to change/modify the generated column-names.

**Comment by murat-cetinkaya** [comment]
Hi. 
I have the case that @rahul-dt mentioned (trying to switch from Fivetran to Airbyte). The thing is that the only thing that Airbyte shouldn't do to solve the problem is to not change the original case of the objects and attributes names. 

In my case, my source is Zuora and the object/attribute names are in camel case (e.g. `accountNumber`). Fivetran change it to snake case (`account_number`) and we use those fields in existing dbt models. It'll be possible to do a similar naming conversion using the same rules as Fivetrans' using a dbt macro if Airbyte doesn't convert all the names to lowercase. This trasition would be very smooth.

If it's easier/faster, can Airbyte leave the original source objects/attributes' case as it is?

For reference and more detail of my case, here's the [link](https://discuss.airbyte.io/t/airbyte-converts-the-source-objects-and-attributes-names-to-uppercase/3005/3) for the discussion I opened.

**Comment by grishick** [comment]
This can be handled by platform and be completely transparent to connectors and normalization. Destination and normalization processes simply receive the catalog from the platform.

[a comment by an automated account is not shown]
